### PR TITLE
Sequence Diagram Generation works both for Plant UML and Mermaid UML.

### PR DESCRIPTION
**Pull request by AxelBelin**
Upgrade Plant UML Sequence Diagram generation feature and implement the same algorithm for Mermaid UML.
Add new tests for Sequence diagram Generation (for Plant and Mermaid).

### CODE REVIEW

File: src/main/java/com/github/forax/umldoc/gen/MermaidGenerator.java [hunk 1]
```diff
@@ -119,6 +121,67 @@ private static void generateEnumConstants(List<Field> fields, Writer writer)
     }
   }
 
+  private static String groupKind(Call.Group.Kind groupKind) {
+    return switch (groupKind) {
+      case LOOP -> "loop\n";
+      case OPTIONAL -> "opt\n";
+      case ALTERNATE -> "alt\n";
+      case PARALLEL -> "par\n";
+      default -> "";
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/gen/MermaidGenerator.java:130]: Do not use a default, but list all possible values

File: src/main/java/com/github/forax/umldoc/gen/MermaidGenerator.java [hunk 2]
```diff
@@ -119,6 +121,67 @@ private static void generateEnumConstants(List<Field> fields, Writer writer)
...
+                entities,
+                writer
+        );
+
+        if (currentGroupKind.equals(Call.Group.Kind.ALTERNATE) && nbCalls < group.calls().size()) {
+          writer.append("else\n");
+        }
+      }
+
+      if (call instanceof Call.MethodCall methodCall) {
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/gen/MermaidGenerator.java:172]: you should use if ... else and throw an exception if it's not one of the possible subtypes

File: src/main/java/com/github/forax/umldoc/gen/PlantUmlGenerator.java [hunk 3]
```diff
@@ -122,60 +119,62 @@ private static void addFooter(Writer writer) throws IOException {
...
     return switch (groupKind) {
-      case LOOP -> "loop";
-      case OPTIONAL -> "opt";
-      case ALTERNATE -> "alt";
-      case NONE, PARALLEL -> "group";
+      case LOOP -> "loop\n";
+      case OPTIONAL -> "opt\n";
+      case ALTERNATE -> "alt\n";
+      case PARALLEL -> "group\n";
+      default -> "";
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/gen/PlantUmlGenerator.java:126]: Same here !

File: src/main/java/com/github/forax/umldoc/gen/PlantUmlGenerator.java [hunk 4]
```diff
@@ -122,60 +119,62 @@ private static void addFooter(Writer writer) throws IOException {
...
-        currentEntityName = methodCall.ownerName();
-      } else if (call instanceof Call.Group groupCall) {
-        generateCalls(groupCall, writer);
-      } else {
-        throw new AssertionError();
+        var srcEntityName = methodCall.ownerName();
+        var dstEntityName = ExecutionPathResolver
+                .findEntityFromMethodName(methodCall.name(), entities).type().name();
+        generateMethodCall(srcEntityName, dstEntityName, methodCall.name(), writer);
       }
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/gen/PlantUmlGenerator.java:172]: Same here

**Comment by AxelBelin** [comment]
Fix errors in PlantUmlGenerator and MermaidGenerator classes.

**Pull request merged**